# DSCI100 Individual Plan


Github link: https://github.com/annixu921-creator/Dsci-100-group-project.git

### Importing Data

In [27]:
library(tidyverse)

url = "http://drive.google.com/uc?rxport-download&id=19dtTv9I4hUdTKPBrM1QgI3A0ru68ssds"
players <- read_csv(url)
dim(players); head(players)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] 196   7

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

In [41]:
distinct(players, experience)
distinct(players, subscribe)
distinct(players, gender)
players |> summarize(max(played_hours), min(played_hours), max(Age, na.rm = TRUE), min(Age, na.rm = TRUE))

experience
1 Pro       
2 Veteran   
3 Amateur   
4 Regular   
5 Beginner

subscribe
1  TRUE    
2 FALSE

gender           
1 Male             
2 Female           
3 Non-binary       
4 Prefer not to say
5 Agender          
6 Two-Spirited     
7 Other

max(played_hours) min(played_hours) max(Age, na.rm = TRUE)
1 223.1             0                 58                    
  min(Age, na.rm = TRUE)
1 9

## 1) Data Description

The **`players`** dataset contains **196 observations and 7 variables.**  
    Variables include: 

+ **`experience`**, charactor, experience level of the player, levels include `Beginner`, `Regular`, `Amateur`, `Veteran`, `Pro`. 
+ **`subscribe`**, logical, whether player is subcribed to game-related newsletter, either `True` or `False`.
+ **`hashedEmail`**, charactor, records the hashed Email address of players.
+ **`played_hours`**, double, records the hours of video game played. 
+ **`name`**, charactor, first name of the player.
+ **`gender`**, charactor, gender of the player, include `Male`, `Female`, `Non-Binary`, `Prefer not to say`, `Agender`, `Two-Spirited`, and `Other`. 
+ **`Age`**, double, Age of the player.

### Potential Issues

1. `played_hours` may not reflect true playtime, players could be on the game but afk, away from keyboard.
2. `Age` may not be accurate as people may lie about their birthday, systematic measurement error.
3. unable to decode the emails in `hashedEmail`,  the variable may not be that useful.

## 2) Questions

## 3) Exploratory Data Analysis and Visualization

## 4) Method and Plan